js/main.js

```diff
@@ -566,15 +566,18 @@ window.addEventListener('DOMContentLoaded', event => {
   $goToMainPageFromFavorites.appendChild(goBackFromFavoritesLinkContainer);
 
   // BASIC SEARCH VIEW
-
-  data.basicSearchArray.forEach(element => renderBasicRecipeInfo(element));
+  if (data.basicSearchArray.length > 0) {
+    data.basicSearchArray.forEach(element => renderBasicRecipeInfo(element));
+  }
 
   // DETAILED SEARCH VIEW
-  var detailedRecipeObject = data.detailRecipeObject;
-  var result = renderRecipeDetailes(detailedRecipeObject);
-  $detailedRecipeContainer.appendChild(result);
-  deleteIcon.setAttribute('class', 'hidden');
-  favoriteIcon.classList.remove('hidden');
+  if (data.detailRecipeObject === true) {
+    var detailedRecipeObject = data.detailRecipeObject;
```

Comment:
I'm not sure I understand this part properly. This property is named `detailRecipeObject` but if it `=== true`, then it's a boolean?

I am not clear why this conditional statement works or helps. To me it would make sense to check this instead?

```js
  if (data.detailRecipeObject !== null) {
    var detailedRecipeObject = data.detailRecipeObject;
```

or

```js
  if (typeof data.detailRecipeObject !=='undefined') {
    var detailedRecipeObject = data.detailRecipeObject;
```